tests/unit/controllers/userController.test.js

```diff
@@ -503,4 +505,91 @@ describe('User Controller', () => {
       expect(response.body).toEqual({ message: 'Internal server error' });
     });
   });
+
+  describe('deleteUser', () => {
+    beforeEach(() => {
+      const user = {
+        email: 'email@test.com',
+        password: 'password',
+        roles: ['clinicadmin'],
+      };
+      vi.spyOn(jwt, 'verify').mockReturnValueOnce({
+        userId: 'userId',
+        roles: ['clinicadmin'],
+      });
+      vi.spyOn(User, 'findById').mockResolvedValue(user);
+      vi.spyOn(redisClient, 'exists').mockResolvedValue(true);
+    });
+
+    it('should delete user successfully', async () => {
+      const user = {
+        _id: 'userId',
+        email: 'email@email.com',
+        password: 'password',
+        roles: ['doctor'],
+      };
+
+      vi.spyOn(User, 'findById').mockResolvedValue(user);
+      vi.spyOn(User, 'findOne').mockResolvedValue(user);
+      vi.spyOn(Role, 'find').mockResolvedValue([
+        {
+          role: 'clinicadmin',
+          permissions: [
+            { method: 'delete', onRoles: ['doctor', 'patient', 'himself'] },
+          ],
+        },
+      ]);
+      vi.spyOn(User, 'findByIdAndDelete').mockResolvedValue(user);
+      vi.spyOn(redisClient, 'del').mockResolvedValue(true);
+
+      const response = await request
+        .delete('/users/userId')
+        .set('Cookie', ['token=authToken&refreshToken=refreshToken']);
+
+      expect(response.status).toBe(204);
+    });
+
+    it('should return 404 if user is not found', async () => {
+      const user = {
+        _id: 'userId',
+        email: 'email@email.com',
+        password: 'password',
+        roles: ['doctor'],
+      };
+
+      vi.spyOn(User, 'findById')
+        .mockReturnValueOnce(user)
+        .mockResolvedValue(null);
+      vi.spyOn(User, 'findOne').mockResolvedValue(user);
+      vi.spyOn(Role, 'find').mockResolvedValue([
+        {
+          role: 'clinicadmin',
+          permissions: [
+            { method: 'delete', onRoles: ['doctor', 'patient', 'himself'] },
+          ],
+        },
+      ]);
+
+      const response = await request
+        .delete('/users/userId')
+        .set('Cookie', ['token=authToken&refreshToken=refreshToken']);
+
+      expect(response.status).toBe(404);
+      expect(response.body).toEqual({ message: 'User not found' });
+    });
+
+    it('should return 500 if there is an error deleting user', async () => {
+      const errorMessage = 'Database error';
+      vi.spyOn(User, 'findByIdAndDelete').mockRejectedValue(
+        new Error(errorMessage)
+      );
+
+      const response = await request
+        .delete('/users/userId')
+        .set('Cookie', ['token=authToken&refreshToken=refreshToken']);
+
+      expect(response.status).toBe(500);
+      expect(response.body).toEqual({ message: 'Internal server error' });
+    });
```

Comment:
here the error 500 is not happening in the findByIdAndDelete, but on the middlewares as they are not mocked. I will solve this in a comit